ch4/problem_1.py

```diff
@@ -1,2 +1,9 @@
-def vectorize():
-    pass
\ No newline at end of file
+def vectorize(func):
+    def new_func(*args):
+        result = []
+        for arg in args:
+            result.append(func(arg))
```

Comment:
This would be more pythonic as a list comprehension, like `return [func(arg) for arg in args]`.